run_all_tests.sh

```diff
@@ -7,9 +7,37 @@
 # Fail immediately in case of errors and/or unset variables
 set -eu -o pipefail
 
-# Run all java tests and native unit tests.
-./run_maven_tests.sh
-
-# Run native integration tests that require prepared classpaths for fake classes.
-./run_native_integration_tests.sh --skip-compile
-./run_ejb_app_tests.sh
+# Run rust code checks if CHECK_RUST is true, or java tests if it's not
+if [ "$CHECK_RUST" = true ] 
```

Comment:
@vitvakatu , @atrykush  what if we move this branching code to `.travis/run_travis_job.sh` as its sole purpose is to pick the right steps to run, and it is not supposed to be used interactively, but keep the other tests in the project root?

Currently `run_all_tests.sh` no longer does what it says it does — it may not run any tests but perform toolchain-agnostic checks.